common/src/main/java/com/wynntils/models/containers/PlayerInventoryModel.java

```diff
@@ -42,6 +43,36 @@ public CappedValue getIngredientPouchSlots() {
         return new CappedValue(getUsedIngredientPouchSlots(), MAX_INGREDIENT_POUCH_SLOTS);
     }
 
+    /**
+     * @return -1 if the item is not equipped, otherwise the slot number
+     */
+    public int getEquippedItemSlot(ItemStack itemStack) {
+        int[] accessorySlots = {9, 10, 11, 12};
+        if (McUtils.player().hasContainerOpen()) {
+            // Scale according to server chest size
+            // Eg. 3 row chest size = 27 (ends on i=26 since 0-index), we would get accessory slots {27, 28, 29, 30}
+            int baseSize = McUtils.player().containerMenu.getItems().size();
+            accessorySlots = new int[] {baseSize, baseSize + 1, baseSize + 2, baseSize + 3};
+        }
+
+        for (int slot : accessorySlots) {
+            ItemStack equipped = McUtils.inventory().getItem(slot);
+            if (ItemStack.isSameItem(equipped, itemStack)) {
+                return slot;
+            }
+        }
+        for (int i = 0; i < 4; i++) {
+            ItemStack equipped = McUtils.inventory().armor.get(i);
+            if (ItemStack.isSameItem(equipped, itemStack)) {
+                return i + 5;
+            }
+        }
+        if (ItemStack.isSameItem(McUtils.player().getItemInHand(InteractionHand.MAIN_HAND), itemStack)) {
+            return McUtils.inventory().selected;
+        }
+        return -1;
```

Comment:
Let's add some newlines.

```suggestion
        for (int slot : accessorySlots) {
            ItemStack equipped = McUtils.inventory().getItem(slot);
            if (ItemStack.isSameItem(equipped, itemStack)) {
                return slot;
            }
        }
        
        for (int i = 0; i < 4; i++) {
            ItemStack equipped = McUtils.inventory().armor.get(i);
            if (ItemStack.isSameItem(equipped, itemStack)) {
                return i + 5;
            }
        }
        
        if (ItemStack.isSameItem(McUtils.player().getItemInHand(InteractionHand.MAIN_HAND), itemStack)) {
            return McUtils.inventory().selected;
        }
        
        return -1;
```